neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -94,16 +95,40 @@ public void CalculateFees()
             else
             {
                 long remainder = Gas % d;
-                if (remainder == 0) return;
                 if (remainder > 0)
                     Gas += d - remainder;
-                else
+                else if (remainder < 0)
                     Gas -= remainder;
             }
             using (Snapshot snapshot = Blockchain.Singleton.GetSnapshot())
             {
+                // Compute verification gas
+
+                consumed = 0;
+                foreach (var witness in Witnesses)
+                {
+                    byte[] verification = witness.VerificationScript;
+                    if (verification.Length == 0)
+                    {
+                        verification = snapshot.Contracts.TryGet(Sender)?.Script;
```

Comment:
Erik, we need to disable this, for two reasons:
1) Contracts.get is a state dependent operation
2) The Script logic of this contract would probably (or certainly) use Storage access to grant allowance to the funds.

The best solution here, which is beautiful and simple on Neo3, is that contract owner can withdraw funds on Application trigger. It could invoke "withdraw" method, which will call a Payable operation to the desired destination. This is much better than allowing a contract invoke here.

Reply:
Do you agree @shargon ? If I'm mistaken feel free to revert my commit.

Reply:
You can explicitly put the contract script in `VerificationScript`. What is the difference?

Reply:
Reverted

Reply:
If you explicitly put the contract there, you won't have access to StorageContext, correct? It's different Erik, this is not a good feature anymore... not without global utxo.

Reply:
> If you explicitly put the contract there, you won't have access to StorageContext, correct?

Incorrect.